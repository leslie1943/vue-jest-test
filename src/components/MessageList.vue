<template>
  <ul>
    <li v-for="(message, index) in messages" :key="index">
      {{ message }}
    </li>
  </ul>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

@Component
export default class MessageList extends Vue {
  @Prop({ required: true }) private messages!: string[]
}
</script>
