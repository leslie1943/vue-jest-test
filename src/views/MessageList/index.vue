<template>
  <div class="home">
    <MessageList :messages="messages" />
    <v-divider />
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import MessageList from '@/components/MessageList.vue'
import { Pet } from '@/types/advanced-readonly.types'

@Component({
  components: {
    MessageList,
  },
  beforeRouteLeave(to, from, next) {
    console.info('Message list beforeRouteLeave >>>>>>>>> to', to)
    console.info('Message list beforeRouteLeave >>>>>>>>> from', from)
    if (to.path !== '/hello-world') {
      delete from.meta.position
    }
    next()
  },
})
export default class MessageListIndex extends Vue {
  messages: string[] = ['Dora', 'Mark', 'Justin']
  mounted(): void {
    this.onReadOnly()
  }

  onReadOnly(): void {
    const pet: Readonly<Pet> = {
      cat: {
        name: 'tom',
        age: 2,
      },
      dog: {
        weight: 22,
        height: 22,
      },
    }
    console.log(
      'ðŸš€ ~ file: index.vue ~ line 30 ~ MessageListIndex ~ mounted ~ pet',
      pet
    )
    // pet.cat = { name: 'jerry', age: 10 } // Cannot assign to 'cat' because it is a read-only property
    pet.cat.name = 'Jerry' // you can chagne
  }
}
</script>
