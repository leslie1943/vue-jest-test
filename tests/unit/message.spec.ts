import { shallowMount } from '@vue/test-utils'
import MessageList from '@/components/MessageList.vue'

describe('MessageList.vue', () => {
  let wrapper: any
  beforeEach(() => {
    wrapper = shallowMount(MessageList, {
      propsData: {
        messages: ['Mark', 'Dora'],
      },
    })
  })
  it('equals messages to ["Dora"]', () => {
    expect((wrapper.vm as any).messages).toEqual(['Mark', 'Dora'])
  })
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
