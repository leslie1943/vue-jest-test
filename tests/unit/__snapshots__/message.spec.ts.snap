// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList.vue has the expected html structure 1`] = `
<ul>
  <li>
    
    Mark
  
  </li>
  <li>
    
    Dora
  
  </li>
</ul>
`;
